src/_guides/language/numbers.md

```diff
@@ -0,0 +1,547 @@
+
+When arithmetic results differ, as described in this section,
+the behavior is **platform specific**
+and **subject to change**.
+
+{{ site.alert.note }}
+  The platform-specific behavior might change to be
+  less surprising, more consistent, or more performant.
+{{ site.alert.end }}
+
+
+### Precision
+
+The following table demonstrates how some numerical expressions
+differ due to precision.
+Here, `math` represents the `dart:math` library,
+and `math.pow(2, 53)` is 2<sup>53</sup>.
+
+On the web, integers lose precision past 53 bits.
+In particular, 2<sup>53</sup> and 2<sup>53</sup>+1
+map to the same value due to truncation.
+***[QUESTION: The language tour's
+[numbers section](https://dart.dev/guides/language/language-tour#numbers)
+says 2<sup>53</sup> - 1, not 2<sup>53</sup>. Should I update it?]***
+On native, these values can still be differentiated
+because native numbers have 64 bits — 63 bits for the value and 1 for the sign.
+
+The effect of overflow is visible
+when comparing 2<sup>63</sup>-1 to 2<sup>63</sup>.
+On native, the latter overflows to -2<sup>63</sup>,
+as expected for two's-complement arithmetic.
+On the web, these values do not overflow
+because they are represented differently;
+they're approximations due to the loss of precision.
+
+<table class="table table-striped nowrap">
+  <tr>
+   <th>Expression</th>
+   <th>Native</th>
+   <th>Web</th>
+  </tr>
+  <tr>
+   <td><code>math.pow(2, 53) - 1</code></td>
+   <td><code>9007199254740991</code></td>
+   <td><code>9007199254740991</code></td>
+  </tr>
+  <tr>
+   <td><code>math.pow(2, 53)</code></td>
+   <td><code>9007199254740992</code></td>
+   <td><code>9007199254740992</code></td>
+  </tr>
+  <tr>
+   <td><code>math.pow(2, 53) + 1</code></td>
+   <td><code>9007199254740993</code></td>
+   <td><code>9007199254740992</code></td>
+  </tr>
+  <tr>
+   <td><code>math.pow(2, 62)</code></td>
+   <td><code>4611686018427387904</code></td>
+   <td><code>4611686018427388000</code></td>
+  </tr>
+  <tr>
+   <td><code>math.pow(2, 63) - 1</code></td>
+   <td><code>9223372036854775807</code></td>
+   <td><code>9223372036854776000</code></td>
+  </tr>
+  <tr>
+   <td><code>math.pow(2, 63)</code></td>
+   <td><code>-9223372036854775808</code></td>
+   <td><code>9223372036854776000</code></td>
+  </tr>
+  <tr>
+   <td><code>math.pow(2, 64)</code></td>
+   <td><code>0</code></td>
+   <td><code>18446744073709552000</code></td>
+  </tr>
+</table>
+
+
+### Identity
+
+On native platforms, `double` and `int` are distinct types:
+no value can be both a `double` and an `int` at the same time.
+On the web, that isn't true.
+Because of this difference,
+identity can differ between platforms,
+although equality (`==`) doesn't.
+***[QUESTION: I changed "shouldn't" to "doesn't", because
+"should" is a red flag for imprecise/incorrect text.
+I considered adding "(for numbers between -2<sup>53</sup> and 2<sup>53</sup>)",
+but that doesn't seem to be true. Did I get it right?
```

Comment:
I think the text is good as written now.

There are values that can be represented on web (e.g., 2^80), but not native and vice versa, so we could spell that out more I suppose.